games/loveletter/rules.js

```diff
@@ -0,0 +1,371 @@
+  if (action.target) {
+    // Action cannot be no-target if someone isn't handmaided
+    if (action.target === -1 && !all_others_handmaided) return false;
+    // Target must be valid, alive, and not handmaided
+    if (!(
+      action.target >= 0
+      && action.target < game_view.num_players
+      && game_view.player_alive[action.target]
+      && !is_handmaided(game_view.discards[action.target])
+    )) return false;
+  }
+  if (action.guess) {
+    // Guess must be "no-guess" or on a non-guard
+    if (!(action.guess === -1 || (action.guess >= 2 && action.guess <= 8))) return false;
+  }
+  if (game_view.hands[game_view.player_id].includes(7)) {
+    // Presence of a countess means you can't play guard or king
+    if (action.card === 5 || action.card === 6) return false;
+  }
+  return true;
+}
+
+function winners(game_state) {
+  const required_wins = num_wins_required(game_state.num_players);
+  const result = [];
+  for (let i = 0; i < game_state.num_players; i += 1) {
+    if (game_state.num_wins[i] >= required_wins) result.push(i);
+  }
+  return result;
+}
+
+function is_game_finished(game_state) {
+  return winners(game_state).length > 0;
+}
+
+function play_card(game_state, player_id, action) {
+  // Mutates game_state.
+  // Returns: true if we should advance the turn.
+  /* eslint-disable no-param-reassign */
+
+  // Discard the chosen card.
+  const my_discards = game_state.discards[player_id];
+  const my_hand = game_state.hands[player_id];
+  my_discards.push(my_hand.splice(my_hand.indexOf(action.card), 1)[0]);
+
+  switch (action.card) {
+    case 1:
+      if (action.target === -1) {
+        game_state.log.push(`Player ${player_id} discards a 1.`);
+        break;
+      }
+      if (game_state.hands[action.target].includes(action.guess)) {
+        // Guess was correct. Fuck 'em up, boss!
+        game_state.log.push(
+          `Player ${player_id} guesses correctly ` +
+          `that Player ${action.target} has a ${action.guess}.`);
+        kill_player(game_state, action.target);
+      } else {
+        game_state.log.push(
+          `Player ${player_id} guesses incorrectly ` +
+          `that Player ${action.target} has a ${action.guess}.`);
+      }
+      break;
+    case 2:
+      // Look at another player's hand.
+      game_state.priested_player = action.target;
+      game_state.log.push(
+        `Player ${player_id} uses a 2 to see Player ${action.target}'s hand.`);
+      // Does not advance the turn. Continued at unpriest().
+      return false;
+    case 3: {
+      // Compare hands; lower loses.
+      const my_card = game_state.hands[player_id][0];
+      const their_card = game_state.hands[action.target][0];
+      if (my_card > their_card) {
+        kill_player(game_state, action.target);
+        game_state.log.push(
+          `Player ${player_id} uses a 3 to compare hands with Player ${action.target}. ` +
+          `Player ${player_id} is defeated.`);
```

Comment:
Backwards. If my_card > their_card, action.target is defeated.

Reply:
I think it's just the message that's backwards. Fixed.